opsdroid/events.py

```diff
@@ -115,3 +123,13 @@ def __init__(self, text, user, room, connector, raw_message=None):
         if 'thinking-delay' in self.connector.configuration:
             await self._thinking_delay()
         return await self.connector.react(self, emoji)
+
+
+class Image(Message):
+    def __init__(self, image_url, user, room, connector):
+        self.created = datetime.now()
+        self.url = image_url
```

Comment:
We need to think carefully about this. Some connectors will want a url, some will want the image itself probably as a binary or base64 encoding.

Reply:
It would be useful to have a method on the image connector to download a URL. i.e. convert a URL to a bytes string, would save all the connectors implementing it themselves.

Reply:
Would that not be an issue for the connector to deal with? I feel like the most sensible thing for the event to do is use a single general representation of the image and then each connector will have to decide the best way to use that, which could include going to the URL and downloading the image.

Reply:
I can think of many examples where there is no url. What if the bot generates a plot with matplotlib for example? Or a QR code with [qrcode](https://pypi.org/project/qrcode/). These will start as image objects in Python.

Reply:
Having discussed this with @SolarDrew today, I think the base representation for a `File` should be a bytes object, this is the only way we can always represent an image.

I think that you should be able to construct an image with either a URL or a bytes object, and then the URL representation and the bytes representation should be accessible as properties, with the understanding that the URL can be `None`.

Reply:
This definitely highlights one of the challenges of opsdroid. It is trying to support "all" chat clients, and so we are going to get issues like this. 

I would be happy for url to default to `None` and have a property for the bytes object which can download the url if accessed. This would result in clients which only accept urls not be able to take bytes only images. But I think thats fine, I see no solution without hosting an image CDN 😆.

Reply:
Gitter?

Reply:
Really??? 🤦🏾‍♀️

Reply:
Exasperation at Gitter aside, opsdroid can only abstract away the differences between chat platforms so far. I think this is just the limit of how far we can go?